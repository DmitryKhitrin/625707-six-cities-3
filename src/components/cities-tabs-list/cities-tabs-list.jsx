import React, {memo} from "react";
import {PropTypes} from "prop-types";
import {CityTab} from "../city-tab/city-tab.jsx";

const CitiesTabsList = ({locations = [], setCity, activeCity}) => {
  return (
    <div className="cities tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {locations.map((city, i) => {
            const {name} = city;
            return (
              <CityTab
                key={i}
                cityName={name}
                setCity={setCity}
                isActive={activeCity === name}
              />
            );
          })}
        </ul>
      </section>
    </div>
  );
};

const MemoizedCitiesTabsList = memo(CitiesTabsList);
export {MemoizedCitiesTabsList as CitiesTabsList};


CitiesTabsList.propTypes = {
  locations: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        cityName: PropTypes.string,
        location: PropTypes.any
      })
  ).isRequired,
  setCity: PropTypes.func.isRequired,
  activeCity: PropTypes.string.isRequired,
};
