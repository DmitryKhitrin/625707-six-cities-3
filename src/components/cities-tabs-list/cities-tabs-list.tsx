import React, {FC, memo} from "react";
import {CityTab} from "../city-tab/city-tab";

type Location = {
  id: string;
  name: string;
  location: number[];
}

type Props = {
  setCity: (city: string) => void;
  activeCity: string;
  locations: Location[]
}

const CitiesTabsList: FC<Props> = ({locations = [], setCity, activeCity}) => {
  return (
    <div className="cities tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {locations.map((city, i) => {
            const {name} = city;
            return (
              <CityTab
                key={i}
                cityName={name}
                setCity={setCity}
                isActive={activeCity === name}
              />
            );
          })}
        </ul>
      </section>
    </div>
  );
};

const MemoizedCitiesTabsList = memo(CitiesTabsList);
export {MemoizedCitiesTabsList as CitiesTabsList};
