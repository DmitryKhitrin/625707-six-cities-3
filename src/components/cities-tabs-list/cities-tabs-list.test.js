import React from "react";
import renderer from "react-test-renderer";
import {CitiesTabsList} from "./cities-tabs-list.jsx";

const locations = [
  {
    id: `1`,
    cityName: `Paris`,
    location: [48.864716, 2.349014]
  },
  {
    id: `2`,
    cityName: `Dusseldorf`,
    location: [51.22172, 6.77616]
  },
];
it(`<CitiesTabsList /> should render CitiesTabsList.`, () => {
  const tree = renderer
    .create(
        <CitiesTabsList
          locations={locations}
          setCity={jest.fn()}
          activeCity={'Moscow'}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
