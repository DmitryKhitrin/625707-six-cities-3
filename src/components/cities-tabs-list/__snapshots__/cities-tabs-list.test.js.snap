// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CitiesTabsList /> should render CitiesTabsList. 1`] = `
<div
  className="cities tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
        >
          <span />
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
        >
          <span />
        </a>
      </li>
    </ul>
  </section>
</div>
`;
